app/src/main/java/com/smashingboxes/epa_prototype_android/MainActivity.java

```diff
@@ -178,7 +185,22 @@ protected void onCreate(Bundle savedInstanceState) {
         epaRequestManager = new EpaRequestManager(this, this);
 
         initializeSwipeHandling();
-        fetchData();
+
+        if(savedInstanceState != null && savedInstanceState.containsKey(KEY_EXTRA_AIR_QUALITY_LIST) && savedInstanceState.containsKey(KEY_EXTRA_TIME_SERIES_LIST)) {
```

Comment:
For the future, it might be better to split up these two key checks so that if the saved instance state doesn't contain `KEY_EXTRA_AIR_QUALITY_LIST`, but it does contain the other key, then at least the other array list is retrieved.
But, maybe this case would never happen, in which case you can just ignore this.